pkg/performanceprofile/cmd/render/render.go

```diff
@@ -210,6 +220,47 @@ func render(inputDir, outputDir string) error {
 	return nil
 }
 
+// isLegacySNOWorkloadPinningMethod provides a check for situations where the user is creating an SNO cluster with the
+// legacy method for CPU Partitioning. In order to make sure the bootstrap and running cluster MCs are synced up we check the MCs
+// provided by the user, if any one of them contain the file addition to `/etc/kubernetes/openshift-workload-pinning` and have not set
+// the API CPUPartitioningAllNodes at install time, we assume a legacy intention and alter the flag to generate the new bootstrap manifests.
+//
+// Note:
+//   - This will only trigger when Control plane is SNO or the CPU Partitioning API is NOT set to AllNodes
+//   - We do not alter the API flag here, when NTO starts up in cluster, it will notice the flag and
+//     update the flag and ignore the create error since the files already exist.
+func isLegacySNOWorkloadPinningMethod(mcs []*mcfgv1.MachineConfig, infra *apicfgv1.Infrastructure, partitioningMode *apicfgv1.CPUPartitioningMode) bool {
```

Comment:
can we add unit tests to this function?
as stretch goal, render e2e tests would be nicer, possibly using realistic manifests

Reply:
Actually, let's think about _A_ form of e2e test coverage and let's try hard to add it.

Reply:
Sounds good, adding them now